src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -301,68 +317,61 @@ private static string DecodePropertyName(string name, string language)
             return name;
         }
 
-        private class DeclarationGenerator : SymbolVisitor
+        /// <summary>
+        /// The <see langword="bool"/> type parameter indicates if we succeeded at writing out the declaration ID or not. 
+        /// </summary>
+        private sealed class PrefixAndDeclarationGenerator(StringBuilder builder) : SymbolVisitor<bool>
         {
-            private readonly StringBuilder _builder;
-            private readonly Generator _generator;
-
-            public DeclarationGenerator(StringBuilder builder)
```

Comment:
>public DeclarationGenerator(StringBuilder builder) [](http://example.com/codeflow?start=12&length=50)

The switch to primary constructor is unnecessary and it makes the code harder to follow. Please revert.

Reply:
BTW, apart from issues specifically with the primary constructor (I would ask to not use it even if that was used with a new type, probably with a very small exception), this also constitutes as a style change. Style changes are not welcome.

Reply:
To me, this is not a style only change.  It fundamentally distinguishes to me a major difference in how an object works, amking it much easier to reason about.  In this case, i can tell that the builder is no longer a field of the generator, it is simply a parameter, and can be thought of as having captured semantics for the purposes of this helper type.  That makes reasoning about how this works much simpler, and more in line with a helper method/function/lambda.

To me, when classes are introduced for this sort of purpose, this is a much easier and simpler way to reason about things, and it (and other changes) helped reveal several problems in this area that had previously gone unnoticed.  I would prefer to keep this.

Reply:
Moved to cleanup pr.